### PR TITLE
test: add more tests for each template option

### PR DESCRIPTION
**Pull request by daniel-makerx**

### CODE REVIEW

File: tests/test_templates.py [hunk 1]
```diff
@@ -1,26 +1,58 @@
...
+        working_generated_root = working_dir / generated_folder
+        shutil.copytree(root, working_dir)
+        subprocess.run(["git", "add", "-A"], cwd=working_dir)
+        subprocess.run(
+            ["git", "commit", "-m", "draft changes", "--no-verify"], cwd=working_dir
+        )
+
+        yield working_dir
+
+        for dir_name in os.listdir(working_generated_root):
```
Comment (achidlow) [inline tests/test_templates.py:32]: https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module
can use `.iterdir()` here

File: tests/test_templates.py [hunk 2]
```diff
@@ -1,26 +1,58 @@
...
+    with tempfile.TemporaryDirectory(ignore_cleanup_errors=True) as temp:
+        working_dir = Path(temp) / "template"
+        working_generated_root = working_dir / generated_folder
+        shutil.copytree(root, working_dir)
+        subprocess.run(["git", "add", "-A"], cwd=working_dir)
+        subprocess.run(
+            ["git", "commit", "-m", "draft changes", "--no-verify"], cwd=working_dir
+        )
+
+        yield working_dir
```
Comment (achidlow) [inline tests/test_templates.py:30]: should this be a try/finally with cleanup below?

Comment (achidlow) [reply tests/test_templates.py:30]: nope, never mind

File: tests/test_templates.py (hunk 1 above)
Comment (daniel-makerx) [reply tests/test_templates.py]: Done

**Pull request merged**